packages/components/react/src/Panels/WorkspacePanel.tsx

```diff
@@ -21,41 +17,80 @@ interface Props {
   theme: Theme;
 }
 
+interface PanelProps extends Props {
+  hasEditor: boolean;
+  hasPreviews: boolean;
+  hideTerminalPanel: boolean;
+}
+
+interface TerminalProps extends PanelProps {
+  terminalPanelRef: React.RefObject<ImperativePanelHandle>;
+  terminalExpanded: React.MutableRefObject<boolean>;
+}
+
 /**
  * This component is the orchestrator between various interactive components.
  */
 export function WorkspacePanel({ tutorialStore, theme }: Props) {
-  const fileTree = tutorialStore.hasFileTree();
   const hasEditor = tutorialStore.hasEditor();
   const hasPreviews = tutorialStore.hasPreviews();
   const hideTerminalPanel = !tutorialStore.hasTerminalPanel();
 
-  const editorPanelRef = useRef<ImperativePanelHandle>(null);
-  const previewPanelRef = useRef<ImperativePanelHandle>(null);
   const terminalPanelRef = useRef<ImperativePanelHandle>(null);
-  const previewRef = useRef<ImperativePreviewHandle>(null);
   const terminalExpanded = useRef(false);
 
-  const [helpAction, setHelpAction] = useState<'solve' | 'reset'>('reset');
+  return (
+    <PanelGroup className={resizePanelStyles.PanelGroup} direction="vertical">
+      <Editor
+        theme={theme}
+        tutorialStore={tutorialStore}
+        hasEditor={hasEditor}
+        hasPreviews={hasPreviews}
+        hideTerminalPanel={hideTerminalPanel}
+      />
 
-  const selectedFile = useStore(tutorialStore.selectedFile);
-  const currentDocument = useStore(tutorialStore.currentDocument);
+      <PanelResizeHandle
+        className={resizePanelStyles.PanelResizeHandle}
+        hitAreaMargins={{ fine: 5, coarse: 5 }}
+        disabled={!hasEditor}
+      />
 
-  const lesson = tutorialStore.lesson!;
+      <Preview
+        theme={theme}
+        tutorialStore={tutorialStore}
+        terminalPanelRef={terminalPanelRef}
+        terminalExpanded={terminalExpanded}
+        hideTerminalPanel={hideTerminalPanel}
+        hasPreviews={hasPreviews}
+        hasEditor={hasEditor}
+      />
 
-  const onEditorChange = useCallback<OnEditorChange>((update) => {
-    tutorialStore.setCurrentDocumentContent(update.content);
-  }, []);
+      <PanelResizeHandle
+        className={resizePanelStyles.PanelResizeHandle}
+        hitAreaMargins={{ fine: 5, coarse: 5 }}
+        disabled={hideTerminalPanel || !hasPreviews}
+      />
 
-  const onEditorScroll = useCallback<OnEditorScroll>((position) => {
-    tutorialStore.setCurrentDocumentScrollPosition(position);
-  }, []);
+      <Terminal
+        tutorialStore={tutorialStore}
+        theme={theme}
+        terminalPanelRef={terminalPanelRef}
+        terminalExpanded={terminalExpanded}
+        hideTerminalPanel={hideTerminalPanel}
+        hasEditor={hasEditor}
+        hasPreviews={hasPreviews}
+      />
+    </PanelGroup>
+  );
+}
 
-  const onFileSelect = useCallback((filePath: string | undefined) => {
-    tutorialStore.setSelectedFile(filePath);
-  }, []);
+function Editor({ theme, tutorialStore, hasEditor }: PanelProps) {
```

Comment:
Should we maybe name this `EditorSection`? My brain has a hard time when multiple components in the same codebase have the same name and are wrappers of the other :see_no_evil:

Reply:
Sure! For consistency I'll rename these 3 internal components to `EditorSection`, `PreviewSection` and `TerminalSection`.